File: cyanBat/src/main/java/at/msmiech/cyanbat/gameobjects/PixmapGameObject.java
package at.msmiech.cyanbat.gameobjects;

import java.util.List;

import android.graphics.Rect;
import at.grueneis.game.framework.Graphics;
import at.grueneis.game.framework.Input.TouchEvent;
import at.grueneis.game.framework.Pixmap;

public abstract class PixmapGameObject extends MovableGameObject {

	public Pixmap pixmap;

	protected PixmapGameObject(Rect rect, Pixmap pixmap) {
		super(rect);
		this.pixmap = pixmap;
	}

	@Override
	public void update(float deltaTime, List<TouchEvent> touchEvents) {
		super.update(deltaTime, touchEvents);
	}

	@Override
	public void draw(Graphics g) {
		super.draw(g);
	}
}

File: cyanBat/src/main/java/at/smiech/cyanbat/gameobjects/PixmapGameObject.kt
package at.smiech.cyanbat.gameobjects

import android.graphics.Rect
import at.grueneis.game.framework.Graphics
import at.grueneis.game.framework.Input.TouchEvent
import at.grueneis.game.framework.Pixmap

abstract class PixmapGameObject protected constructor(rect: Rect, var pixmap: Pixmap) : MovableGameObject(rect)
